core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -56,14 +56,14 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   }
 
   /** Set a configuration variable. */
-  def set(key: String, value: String): SparkConf = {
+  def set(key: String, value: Any): SparkConf = {
```

Comment:
I recompiled spark with this change and played around with spark-shell and everything seems to be fine.